core/src/main/java/feast/core/controller/CoreServiceRestController.java

```diff
@@ -0,0 +1,159 @@
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsRequest;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsRequest.Builder;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsRequest;
+import feast.proto.core.CoreServiceProto.ListFeaturesRequest;
+import feast.proto.core.CoreServiceProto.ListProjectsResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@CrossOrigin
+@RequestMapping(value = "/api/v1", produces = "application/json")
+public class CoreServiceRestController {
+
+  private final JsonFormat.Printer jsonPrinter;
+  private final FeastProperties feastProperties;
+  private SpecService specService;
+  private JobService jobService;
+  private StatsService statsService;
+  private AccessManagementService accessManagementService;
+
+  @Autowired
+  public CoreServiceRestController(
+      FeastProperties feastProperties,
+      SpecService specService,
+      JobService jobService,
+      StatsService statsService,
+      AccessManagementService accessManagementService) {
+    this.feastProperties = feastProperties;
+    this.specService = specService;
+    this.jobService = jobService;
+    this.statsService = statsService;
+    this.accessManagementService = accessManagementService;
+    this.jsonPrinter = JsonFormat.printer();
+  }
+
+  @RequestMapping(value = "/version", method = RequestMethod.GET)
+  @ResponseBody
+  public String getVersion() throws InvalidProtocolBufferException {
+    GetFeastCoreVersionResponse response =
+        GetFeastCoreVersionResponse.newBuilder().setVersion(feastProperties.getVersion()).build();
+    return jsonPrinter.print(response);
+  }
+
+  @RequestMapping(
+      value = "/project/{project}/feature-set/{featureSetName}",
+      method = RequestMethod.GET)
+  @ResponseBody
+  public String getFeatureSet(@PathVariable String project, @PathVariable String featureSetName)
+      throws InvalidProtocolBufferException {
+    GetFeatureSetRequest request =
+        GetFeatureSetRequest.newBuilder().setProject(project).setName(featureSetName).build();
+    return jsonPrinter.print(specService.getFeatureSet(request));
+  }
+
+  @RequestMapping(value = "/feature-sets", method = RequestMethod.GET)
+  @ResponseBody
+  public String listFeatureSets(
+      @RequestParam String project, @RequestParam(defaultValue = "*") String name)
```

Comment:
Do we have to set a default value here? Wouldn't this cause duplication? Can we try to move the defaults into the services?

Reply:
Yep, will keep http and grpc behaviour consistent on which parameters are optional. Removed the default value here.